core/src/main/java/org/apache/spark/io/ReadAheadInputStream.java

```diff
@@ -247,7 +245,7 @@ public int read() throws IOException {
       // short path - just get one byte.
       return activeBuffer.get() & 0xFF;
     } else {
-      byte[] oneByteArray = oneByte.get();
```

Comment:
The original writing seems to be to avoid creating `byte[1]` repeatedly. Have you encountered any bad cases? The new writing seems to create more objects of type `byte[1]`

Reply:
`byte[1]` is very cheap and it could avoid GC.

Reply:
We can decorate it with `private final` if we really need to avoid creating `byte[1]` repeatedly.